bosh-setup/azuredeploy.json

```diff
@@ -2,149 +2,116 @@
       }
     },
     "clientSecret": {
       "type": "securestring",
-      "defaultValue": "CLIENT-SECRET",
       "metadata": {
         "description": "secret of the client. See https://github.com/cloudfoundry-incubator/bosh-azure-cpi-release/blob/master/docs/guidance.md"
       }
+    },
+    "autoDeployBosh": {
+      "type": "string",
+      "defaultValue": "disabled",
+      "allowedValues": [
+        "enabled",
+        "disabled"
+      ],
+      "metadata": {
+        "description": "The flag allowing to deploy Bosh automatically or not"
+      }
     }
   },
   "variables": {
-    "api-version": "2015-05-01-preview",
+    "apiVersion": "2015-06-15",
     "location": "[resourceGroup().location]",
-    "storageAccountType": "Standard_GRS",
+
+    "defaultStorageAccountName": "[concat('cfdefaultsa', uniqueString(resourceGroup().id))]",
+    "storageAccountType": "Standard_RAGRS",
     "vmStorageAccountContainerName": "vhds",
-    "storageid": "[resourceId('Microsoft.Storage/storageAccounts', parameters('newStorageAccountName'))]",
-    "virtualNetworkName": "[parameters('virtualNetworkName')]",
-    "addressPrefix": "10.0.0.0/16",
+    "storageid": "[resourceId('Microsoft.Storage/storageAccounts', variables('defaultStorageAccountName'))]",
+
+    "virtualNetworkName": "boshvnet-crp",
+    "virtualNetworkAddressSpace": "10.0.0.0/16",
     "vnetID": "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworkName'))]",
-    "subnet1Name": "[parameters('subnetNameForBosh')]",
-    "subnet1Prefix": "10.0.0.0/24",
-    "subnet1Ref": "[concat(variables('vnetID'),'/subnets/', variables('subnet1Name'))]",
-    "subnet2Name": "[parameters('subnetNameForCloudFoundry')]",
-    "subnet2Prefix": "10.0.16.0/24",
+    "subnetNameForBosh": "Bosh",
+    "subnetAddressRangeForBosh": "10.0.0.0/24",
+    "subnetRefForBosh": "[concat(variables('vnetID'),'/subnets/', variables('subnetNameForBosh'))]",
+    "subnetNameForCloudFoundry": "CloudFoundry",
+    "subnetAddressRangeForCloudFoundry": "10.0.16.0/20",
+
+    "vmSize": "Standard_D1",
     "nicName": "[parameters('vmName')]",
     "devboxPrivateIPAddress": "10.0.0.100",
-    "devboxPublicIPAddressID": "[resourceId('Microsoft.Network/publicIPAddresses', concat(parameters('vmName'),'-devbox'))]",
     "imagePublisher": "Canonical",
     "imageOffer": "UbuntuServer",
-    "ubuntuOSVersion": "14.04.2-LTS",
+    "ubuntuOSVersion": "14.04.3-LTS",
+    "sshKeyPath": "[concat('/home/',parameters('adminUsername'),'/.ssh/authorized_keys')]",
+
     "repository": "Azure",
     "branch": "master",
-    "githubPath": "[concat('https://raw.githubusercontent.com/', variables('repository'), '/azure-quickstart-templates/', variables('branch'), '/bosh-setup/')]"
+    "baseUri": "[concat('https://raw.githubusercontent.com/', variables('repository'), '/azure-quickstart-templates/', variables('branch'), '/bosh-setup/')]",
+
```

Comment:
generally, the best approach I've seen is to use a templateBaseUrl parameter, set the default value to `https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/bosh-setup/` and omit this parameter from azuredeploy.json. This let's folks who fork the template or host it somewhere else to host the scripts from their repo by just passing a link to their fork, but still lets the CI to the relinking.

You don't really have to do that since we haven't really put out a best practice/standards on how to provide your base url.